### PR TITLE
Trigger map

### PR DESCRIPTION
**Pull request by godrei**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-io/bitrise/blob/master/.github/CONTRIBUTING.md)
- [x] `README.md` is updated with the changes (if needed)

### Version

Requires a *MAJOR/MINOR/PATCH* [version update](https://semver.org/)

### Context

This PR tidies up trigger map related models and methods to prepare for upcoming changes.

### Changes

- Trigger map related models and methods are moved to `trigger_map.go`, related tests are moved to `trigger_map_test.go`
- Trigger map item related models and methods are moved to `trigger_map_item.go`, related tests are moved to `trigger_map_item_test.go`
- `getPipelineAndWorkflowIDByParams` functionality is moved to `TriggerMapModel.FirstMatchingTarget`
- Trigger map validations moved from BitriseDataModel.Validate to `TriggerMapModel.Validate`
- ~~A trigger map targeting a utility workflow is considered an error instead of a warning~~

### CODE REVIEW

**Comment by ofalvai** [comment]
> A trigger map targeting a utility workflow is considered an error instead of a warning

Is this safe to make this change? Does the website prevent this kind of behavior anyway? I want to avoid breaking customer workflows

File: models/trigger_map.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
+package models
+
+import "fmt"
+
+// TriggerMapModel ...
+type TriggerMapModel []TriggerMapItemModel
+
+// Validate ...
```
Comment (ofalvai) [inline models/trigger_map.go:8]: nit: we can clean up these dummy comments, we are not using golint anymore.

**Comment by godrei** [comment]
@ofalvai regarding this:

> A trigger map targeting a utility workflow is considered an error instead of a warning
Is this safe to make this change? Does the website prevent this kind of behavior anyway? I want to avoid breaking customer workflows

The thing is that you can't select a utility WF in the WFE when configuring a trigger map item:
Screenshot 2024-01-09 at 14 16 15

However, you can set it on the bitrise.yml tab, In this case, the WFE will show a warning:
Screenshot 2024-01-09 at 14 15 44

A build can be triggered, but it fails:
Screenshot 2024-01-09 at 14 17 53

File: models/trigger_map.go (hunk 1 above)
Comment (godrei) [reply models/trigger_map.go]: Fixed here: https://github.com/bitrise-io/bitrise/pull/905/commits/08c5f009771caac0f336d9a28c1eef0eabf114ed

**Comment by godrei** [comment]
@ofalvai I decided to revert the utility workflow targeting warning - error change here: https://github.com/bitrise-io/bitrise/pull/905/commits/11f2e4f276bed99da0f7e34dc9435da8734e6e1e
and also an error message update: https://github.com/bitrise-io/bitrise/pull/905/commits/d3f862fa35b03d54efc2fdc39de2df20d51d456c
to stay on the safe side and created a bug ticket to fix utility workflow targeting.

**Review by ofalvai — APPROVED** [review]

**Pull request merged**